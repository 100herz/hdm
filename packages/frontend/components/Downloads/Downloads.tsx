import { IFiles } from '@/models/files'
import styled from '@emotion/styled'
import dynamic from 'next/dynamic'

interface DownloadsProps extends React.HTMLAttributes<HTMLElement> {
  files: IFiles[]
  title: string
}

export default function Downloads({ files, title, className = '' }: DownloadsProps) {
  const Pdf = dynamic(() => import('@/components/Utils/Pdf'), { ssr: false })

  return (
    <SectionStyled id="kontakt" className={`container lg:px-32 ${className}`}>
      <h3 className="mt-24 text-2xl font-bold lg:text-3xl">{title}</h3>

      <div className="mt-8 download-list">
        {files.map(file => (
          <div key={file.id} className="lg:text-lg download-list__item">
            <Pdf title={file.displayName} filename={file.path} buttonStyleLight buttonWidthFull />
          </div>
        ))}
      </div>
    </SectionStyled>
  )
}

const SectionStyled = styled.section`
  display: block;

  .download-list {
    &__item {
      margin-top: 1rem;
    }

    @media (min-width: 1024px) {
      display: flex;
      flex-wrap: wrap;
      width: 100%;

      &__item {
        flex: 40%;
        margin-top: 2rem;

        &:nth-of-type(even) {
          margin-left: 2rem;
        }
      }
    }
  }
`
