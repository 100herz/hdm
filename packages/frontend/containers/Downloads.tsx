import Downloads from '@/components/Downloads/Downloads'
import Layout from '@/components/Layout'
import certificatesCeDop from '@/data/ce-dop.json'
import certificatesFscPefc from '@/data/fsc-pefc.json'
import { IFiles } from '@/models/files'

export default function DownloadContainer() {
  return (
    <Layout title="Downloads * HDM &ndash; Holz Design Moers">
      <h2 className="container mt-12 text-4xl font-bold lg:mt-32 lg:px-32 lg:text-5xl">Downloads</h2>

      <Downloads title="FSC / PEFC" files={certificatesFscPefc as IFiles[]} />

      <Downloads title="CE-DOP Leistungsbeschreibungen" files={certificatesCeDop as IFiles[]} />
    </Layout>
  )
}
