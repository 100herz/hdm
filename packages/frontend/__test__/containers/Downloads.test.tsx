import Downloads from '@/components/Downloads/Downloads'
import DownloadContainer from '@/containers/Downloads'
import { shallow, ShallowWrapper } from 'enzyme'
import toJSON from 'enzyme-to-json'

describe('IndexPage Testings', () => {
  let wrapper: ShallowWrapper

  beforeEach(() => {
    wrapper = shallow(<DownloadContainer />)
  })

  test('matches snapshot', () => {
    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  test('render one Downloads component', () => {
    expect(wrapper.find(Downloads)).toHaveLength(2)
  })
})
