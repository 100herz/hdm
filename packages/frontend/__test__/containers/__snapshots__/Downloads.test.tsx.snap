// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IndexPage Testings matches snapshot 1`] = `
<Layout
  title="Downloads * HDM â€“ Holz Design Moers"
>
  <h2
    className="container mt-12 text-4xl font-bold lg:mt-32 lg:px-32 lg:text-5xl"
  >
    Downloads
  </h2>
  <Downloads
    files={
      Array [
        Object {
          "displayName": "FSC Zertifikat 2020 - DE",
          "id": "1",
          "path": "downloads/fsc-pefc/hdm-fsc-zertifikat-2020-de.pdf",
        },
        Object {
          "displayName": "FSC Certificate 2020 - EN",
          "id": "2",
          "path": "downloads/fsc-pefc/hdm-fsc-certificate-2020-en.pdf",
        },
        Object {
          "displayName": "PEFC Zertifikat 2020 - DE",
          "id": "3",
          "path": "downloads/fsc-pefc/hdm-pefc-zertifikat-2020-de.pdf",
        },
        Object {
          "displayName": "PEFC Certificate 2020 - EN",
          "id": "4",
          "path": "downloads/fsc-pefc/hdm-pefc-certificate-2020-en.pdf",
        },
      ]
    }
    title="FSC / PEFC"
  />
  <Downloads
    files={
      Array [
        Object {
          "displayName": "DOP-PB202001 - DE",
          "id": "1",
          "path": "downloads/ce-dop/hdm-dop-pb202001-de.pdf",
        },
        Object {
          "displayName": "DOP-PB202001 - EN",
          "id": "2",
          "path": "downloads/ce-dop/hdm-dop-pb202001-en.pdf",
        },
        Object {
          "displayName": "DOP-PG202002 - DE",
          "id": "3",
          "path": "downloads/ce-dop/hdm-dop-pg202002-de.pdf",
        },
        Object {
          "displayName": "DOP-PG202002 - EN",
          "id": "4",
          "path": "downloads/ce-dop/hdm-dop-pg202002-en.pdf",
        },
        Object {
          "displayName": "DOP-PS202003 - DE",
          "id": "5",
          "path": "downloads/ce-dop/hdm-dop-ps202003-de.pdf",
        },
        Object {
          "displayName": "DOP-PS202003 - EN",
          "id": "6",
          "path": "downloads/ce-dop/hdm-dop-ps202003-en.pdf",
        },
        Object {
          "displayName": "DOP-PP202004 - DE",
          "id": "7",
          "path": "downloads/ce-dop/hdm-dop-pp202004-de.pdf",
        },
        Object {
          "displayName": "DOP-PP202004 - EN",
          "id": "8",
          "path": "downloads/ce-dop/hdm-dop-pp202004-en.pdf",
        },
      ]
    }
    title="CE-DOP Leistungsbeschreibungen"
  />
</Layout>
`;
